Sources/StreamChat/APIClient/Endpoints/Payloads/IdentifiablePayload.swift

```diff
@@ -0,0 +1,224 @@
+        channel.fillIds(cache: &cache)
+        watchers?.fillIds(cache: &cache)
+        membership?.fillIds(cache: &cache)
+        messages.fillIds(cache: &cache)
+        pinnedMessages.fillIds(cache: &cache)
+        channelReads.fillIds(cache: &cache)
+    }
+}
+
+extension ChannelDetailPayload: IdentifiablePayload {
+    var databaseId: String? { cid.rawValue }
+    static let modelClass: (NSManagedObject & IdentifiableModel).Type? = ChannelDTO.self
+
+    func fillIds(cache: inout [String: Set<String>]) {
+        addId(cache: &cache)
+        createdBy?.fillIds(cache: &cache)
+        members?.fillIds(cache: &cache)
+        invitedMembers.fillIds(cache: &cache)
+    }
+}
+
+extension UserPayload: IdentifiablePayload {
+    var databaseId: String? { id }
+    static let modelClass: (NSManagedObject & IdentifiableModel).Type? = UserDTO.self
+
+    func fillIds(cache: inout [String: Set<String>]) {
+        addId(cache: &cache)
+    }
+}
+
+extension MessagePayload: IdentifiablePayload {
+    var databaseId: String? { id }
+    static let modelClass: (NSManagedObject & IdentifiableModel).Type? = MessageDTO.self
+
+    func fillIds(cache: inout [String: Set<String>]) {
+        addId(cache: &cache)
+        user.fillIds(cache: &cache)
+        quotedMessage?.fillIds(cache: &cache)
+        mentionedUsers.fillIds(cache: &cache)
+        threadParticipants.fillIds(cache: &cache)
+        latestReactions.fillIds(cache: &cache)
+        ownReactions.fillIds(cache: &cache)
+        pinnedBy?.fillIds(cache: &cache)
+        pinnedBy?.fillIds(cache: &cache)
+    }
+}
+
+extension MessageReactionPayload: IdentifiablePayload {
+    var databaseId: String? {
+        MessageReactionDTO.createId(userId: user.id, messageId: messageId, type: type)
+    }
+
+    static let modelClass: (NSManagedObject & IdentifiableModel).Type? = MessageReactionDTO.self
+
+    func fillIds(cache: inout [String: Set<String>]) {
+        addId(cache: &cache)
+        user.fillIds(cache: &cache)
+    }
+}
+
+extension MemberPayload: IdentifiablePayload {
+    var databaseId: String? { nil } // Cannot build id without channel id
+    static let modelClass: (NSManagedObject & IdentifiableModel).Type? = MemberDTO.self
+
+    func fillIds(cache: inout [String: Set<String>]) {
+        addId(cache: &cache)
+        user.fillIds(cache: &cache)
+    }
+}
+
+extension ChannelReadPayload: IdentifiablePayload {
+    var databaseId: String? { nil } // Needs a composed predicate 'channel.cid == %@ && user.id == %@'
+    static let modelClass: (NSManagedObject & IdentifiableModel).Type? = ChannelReadDTO.self
+
+    func fillIds(cache: inout [String: Set<String>]) {
+        addId(cache: &cache)
+        user.fillIds(cache: &cache)
+    }
+}
+
+private extension NSManagedObject {
+    static func batchFetch(keyPath: String, equalTo values: [String], context: NSManagedObjectContext) -> [NSManagedObject] {
+        let request = NSFetchRequest<Self>(entityName: entityName)
```

Comment:
You don't need to fetch all the objects, you can set
```
request = .managedObjectIDResultType
```
and then CoreData will only return ManagedObjectIds, resulting in better performance

Reply:
You can do that, but then you need to get the results in the same order to be able to match id to objectID. Which doesn't seem to be the case.

Reply:
I now realized the snippet i've pasted was missing some important characters :D 

```
request.resultType = .managedObjectIDResultType
```

this won't change the ordering or anything, it'll just make sure only objectIDs are returned

Reply:
Yeah, but that is not enough. We need to create a mapping between id and objectID. If we only get the objectID, we don't know to which id it belongs